### ISSUE TITLE
Language plugins: invalid `language` id error does not enumerate the registered language keys

### ISSUE THREAD

**Issue by mike-north** [eslint/eslint#21171]
# Language plugins: invalid `language` id error doesn't enumerate the registered language keys

## What problem does this feature solve?

When a flat-config object references a plugin's `languages` namespace with a `language` key
that doesn't exist on that plugin (a typo, e.g. `language: "sh/shell"` instead of the
registered `language: "sh/sh"`), `Config`'s constructor throws:

```
TypeError: Key "language": Could not find "shell" in plugin "sh".
```

(`lib/config/config.js`, `Config` constructor — the check right after `splitPluginIdentifier`.)

This message names the offending key but never lists what *would* have worked — the plugin
object's own `languages` keys are already in hand at the point the error is thrown
(`plugins[languagePluginName].languages`), but they're not surfaced. A user (or an author
following a plugin's own docs) has to go read the plugin's source/docs to locate the correct
`language` id instead of getting it from the error itself.

Reproduction (`eslint` 10.7.0, a real language-plugin package registering `languages: { sh:
ShellLanguage }`):

```js
import { Linter } from 'eslint';
import sh from '@eslint-sh/eslint-plugin';

new Linter().verify('echo hi\n', [
  { files: ['**/*.sh'], plugins: { sh }, language: 'sh/nope' },
], 'a.sh');
// TypeError: Key "language": Could not find "nope" in plugin "sh".
```

## What did you expect to happen?

Something closer to how the sibling `Key "languageOptions": Unexpected key "..." found.`
validation error behaves elsewhere in the same file (and how enum-style option errors read
throughout the codebase): name the value that was actually available. For example:

```
TypeError: Key "language": Could not find "nope" in plugin "sh". Available languages in
plugin "sh": "sh".
```

## What did you see instead?

```
TypeError: Key "language": Could not find "nope" in plugin "sh".
```

No enumeration of the plugin's actually-registered `languages` keys.

## Are you willing to submit a pull request to fix this?

Possibly — happy to discuss the exact message wording first.

---
We hit this in a DX study for a shell-script language plugin
(https://github.com/mike-north/eslint-sh) — filing this as a companion to #21170 (the related
"config references a plugin's language but omits `language`" diagnostic gap), since both
originate in the same `Config` constructor validation and share the same fix motivation: make
config-time language-selection mistakes self-diagnosing from the error message alone.
